server/apps/immich/src/api-v1/asset/asset.controller.ts

```diff
@@ -103,12 +108,9 @@ export class AssetController {
   async downloadFile(
     @GetAuthUser() authUser: AuthUserDto,
     @Response({ passthrough: true }) res: Res,
-    @Query(new ValidationPipe({ transform: true })) query: ServeFileDto,
     @Param('assetId') assetId: string,
   ): Promise<any> {
-    this.assetService.checkDownloadAccess(authUser);
-    await this.assetService.checkAssetsAccess(authUser, [assetId]);
-    return this.assetService.downloadFile(query, assetId, res);
+    return this.assetService.downloadFile(authUser, assetId).then(asStreamableFile);
```

Comment:
I find this a bit hard to follow. Can you elaborate for documentation?

Reply:
Ah, downloadFile return `ImmichStream` and you cast that to StreamableFile of Nestjs

Reply:
Could return a streamable file directly, but I'd prefer to keep that nestjs reference out of the domain, since it is specific to the api (not cli, graphql, etc.), so I made a separate interface with no nestjs specifics in it (immich readable file), and then map between the two here.